src/merge.rs

```diff
@@ -84,37 +86,35 @@ impl MergeValue {
                 if *height == 0 {
                     *value
                 } else {
-                    self.into_merge_with_zero::<H>().hash::<H>()
+                    into_merge_with_zero::<H>(*key, *value, *height).hash::<H>()
                 }
             }
         }
     }
+}
 
-    /// Helper function for Shortcut node
-    /// Transform it into a MergeWithZero node
-    /// Notice: When tried to call with a MergeValue::Value, this function will panic
-    #[cfg(feature = "trie")]
-    pub fn into_merge_with_zero<H: Hasher + Default>(&self) -> MergeValue {
-        match self {
-            MergeValue::ShortCut { key, value, height } => {
-                let base_key = key.parent_path(0);
-                let base_node = hash_base_node::<H>(0, &base_key, value);
-                let mut zero_bits = *key;
-                for i in *height..=core::u8::MAX {
-                    if key.get_bit(i) {
-                        zero_bits.clear_bit(i);
-                    }
-                }
-                MergeValue::MergeWithZero {
-                    base_node,
-                    zero_bits,
-                    zero_count: *height,
-                }
-            }
-            _ => {
-                panic!("Attempt to convert a MergeValue::Value into MergeValue::ShortCut, which is not supposed to be available.");
+/// Helper function for Shortcut node
+/// Transform it into a MergeWithZero node
+/// Notice: When tried to call with a MergeValue::Value, this function will panic
```

Comment:
suggest to reword comment and rename the helper fn
```suggestion
/// Helper function for Shortcut node
/// Transform it into a MergeValue or MergeWithZero node
#[cfg(feature = "trie")]
pub fn into_merge_value<H: Hasher + Default>(key: H256, value: H256, height: u8) -> MergeValue {
```